### PR TITLE
Add `MockBackend` tests

### PR DESCRIPTION
**Pull request by georgwiese**
Adds some (negative) tests for `MockBackend`.

### CODE REVIEW

File: pipeline/tests/pil.rs [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#[cfg(feature = "halo2")]
```
Comment (georgwiese) [inline pipeline/tests/pil.rs:1]: I think that was just wrong? With this, you can't run tests without the `halo2` feature.

Comment (leonardoalt) [reply pipeline/tests/pil.rs:1]: This only applies to the next item, not the whole mod.

File: pipeline/tests/mock_backend.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        col("Fibonacci::x", [1, 2, 3, 5]),
+        col("Fibonacci::y", [2, 3, 5, 8]),
+    ]);
+    test_mock_backend(pipeline);
+}
+
+#[test]
+fn block_to_block_correct_witness() {
+    init_logger();
+    let f = "asm/block_to_block.asm";
```
Comment (leonardoalt) [inline pipeline/tests/mock_backend.rs:45]: Don't we already test this in `asm.rs`?

Comment (georgwiese) [reply pipeline/tests/mock_backend.rs:45]: Yeah, true. Also for the correct fibonacci witness. Should I remove them? I think it's kinda nice to have them side by side.

File: pipeline/tests/pil.rs (hunk 1 above)
Comment (georgwiese) [reply pipeline/tests/pil.rs:1]: Yeah, exactly, but why would we import Goldilocks only if the `halo2` feature is enabled?

File: pipeline/tests/mock_backend.rs (hunk 2 above)
Comment (georgwiese) [reply pipeline/tests/mock_backend.rs]: The other test tests more (more fields & provers). I removed the redundant tests here!

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**